### DISCUSSION TITLE
Deploy Remix app with Express server

### DISCUSSION THREAD

**Discussion by alexandrsashin** [remix-run/remix#6438] [Q&A]
**The problem:**
I have Remix app with Express server and use TypeScript. It works correctly when I develop and run production build using command `cross-env NODE_ENV=production ts-node ./server.ts`.

But on the host I need to use `server.js`. I transpile ts files connected with server.ts and after that try to run `node ./server.js`.
I get the error:
```
/build/index.js:37637
import { PassThrough } from "node:stream";
^^^^^^

SyntaxError: Cannot use import statement outside a module
```

There are some solutions on the Internet, one of them: add `"type": "module"` to _package.json_ and rename server.js to server.cjs, but it is not appropriate for me. Please share different solutions.

**Environment / libraries:**
node v16.16.0
@remix-run/express ^1.16.0
@remix-run/node ^1.16.0
@remix-run/react ^1.16.0

**Comment by kiliman** [comment]
That's odd. `remix build`, which uses `esbuild`, transpiles to CJS by default, so it should have converted the `import` to `require`.

What host are you using? Is it running `npm run build` during deployment?

**Reply by alexandrsashin** [reply]
Maybe the problem in https://github.com/alexandrsashin/remix-app-for-build/blob/master/app/entry.server.tsx#L7 (I modified some lines when worked with mui theme).
In this repository (https://github.com/alexandrsashin/remix-app-for-build) you can try to run production build.

**Reply by alexandrsashin** [reply]
I didn't find a real reason of this problem and recreated project from "zero". And it works.
